### PR TITLE
set pull option default and add to AWXKit

### PR DESCRIPTION
**Pull request by rebeccahhh**
##### SUMMARY

set default option for pulling the container to none, this does not change the 3 available options for this field (which are still `Always`, `Never`, or `Missing`). This does set the default for the model and the DB to None.
This PR also adds the pull field into into the AWXKit api pages model

##### ISSUE TYPE

 - Bughancement Pull Request

##### COMPONENT NAME

 - API

##### AWX VERSION

```
awx: 17.0.1
```

### CODE REVIEW

File: awx/main/models/execution_environments.py [hunk 1]
```diff
@@ -44,7 +44,7 @@ class Meta:
     pull = models.CharField(
         max_length=1024,
         choices=PULL_CHOICES,
-        default='missing',
+        default=None,
```
Comment (shanemcd) [inline awx/main/models/execution_environments.py]: I believe you also want to add 
```        
blank=True,
null=True,
```

See here for another example: https://github.com/ansible/awx/blob/devel/awx/main/models/ha.py#L198-L200

File: awxkit/awxkit/api/pages/execution_environments.py [hunk 2]
```diff
@@ -33,12 +33,13 @@ def create_payload(self, name='', organization=Organization, **kwargs):
...
         return payload
 
-    def payload(self, name='', image=None, organization=None, credential=None, **kwargs):
+    def payload(self, name='', image=None, organization=None, credential=None, pull=None, **kwargs):
         payload = PseudoNamespace(
             name=name or "EE - {}".format(random_title()),
             image=image or random_title(10),
             organization=organization.id if organization else None,
             credential=credential.id if credential else None,
+            pull=pull if pull else None,
```
Comment (shanemcd) [inline awxkit/awxkit/api/pages/execution_environments.py]: I think this can just be `pull=pull` since it _is_ `None` by default.

Comment (rebeccahhh) [reply awxkit/awxkit/api/pages/execution_environments.py]: credential is none by default as well, I just was matching up, but I can change it

Comment (shanemcd) [reply awxkit/awxkit/api/pages/execution_environments.py]: It's passing along the _id_ of the credential if there is one:

```
credential=credential.id if credential else None
```

Here you are just passing along `pull`, so this conditional is unnecessary.

Comment (rebeccahhh) [reply awxkit/awxkit/api/pages/execution_environments.py]: done

**Review by shanemcd — APPROVED** [review]
👍

File: awx/main/models/execution_environments.py [hunk 3]
```diff
@@ -44,7 +44,9 @@ class Meta:
     pull = models.CharField(
         max_length=1024,
```
Comment (jbradberry) [inline awx/main/models/execution_environments.py]: I'd go for something much shorter than this.

File: awx/main/models/execution_environments.py [hunk 4]
```diff
@@ -42,9 +42,11 @@ class Meta:
         on_delete=models.SET_NULL,
     )
     pull = models.CharField(
-        max_length=1024,
+        max_length=16,
         choices=PULL_CHOICES,
-        default='missing',
+        blank=True,
+        null=True,
```
Comment (jbradberry) [inline awx/main/models/execution_environments.py]: We should also ditch `null=True`.

Comment (rebeccahhh) [reply awx/main/models/execution_environments.py]: I'm going to remove that and interactive rebase, we don't need this many commits.

**Review by jbradberry — APPROVED** [review]

**Pull request merged**